crates/re_viewer/src/ui/visible_history.rs

```diff
@@ -164,49 +167,82 @@ pub fn visible_history_ui(
-                    &resolved_visible_history.to,
-                    sequence_timeline,
-                    false,
-                );
-            }
+                    )
+                })
+                .inner;
+        } else {
+            resolved_visible_history_boundary_ui(
+                ctx,
+                ui,
+                &resolved_visible_history.from,
+                sequence_timeline,
+                true,
+            );
+            resolved_visible_history_boundary_ui(
+                ctx,
+                ui,
+                &resolved_visible_history.to,
+                sequence_timeline,
+                false,
+            );
+        }
 
-            current_range_ui(ctx, ui, current_time, sequence_timeline, visible_history);
+        current_range_ui(ctx, ui, current_time, sequence_timeline, visible_history);
 
-            ui.add(
-                egui::Label::new(
-                    egui::RichText::new(if sequence_timeline {
-                        "These settings apply to all sequence timelines."
-                    } else {
-                        "These settings apply to all temporal timelines."
-                    })
-                    .italics()
-                    .weak(),
-                )
-                .wrap(true),
+        ui.add(
+            egui::Label::new(
+                egui::RichText::new(if sequence_timeline {
+                    "These settings apply to all sequence timelines."
+                } else {
+                    "These settings apply to all temporal timelines."
+                })
+                .italics()
+                .weak(),
             )
-            .on_hover_text(
-                "Visible Time Range properties are stored separately for each types of timelines. \
-            They may differ depending on whether the current timeline is temporal or a sequence.",
-            );
-        })
-        .header_response
+            .wrap(true),
+        )
         .on_hover_text(
-            "Controls the time range used to display data in the Space View.\n\n\
-        Note that the data immediately preceding the time range is always displayed.",
+            "Visible Time Range properties are stored separately for each types of timelines. \
+            They may differ depending on whether the current timeline is temporal or a sequence.",
         );
+    });
+
+    // Decide when to show the visible history highlight in the timeline. The trick is that when
+    // interacting with the controls, the mouse might end up outside the collapsing header rect,
+    // so we must track these interactions specifically.
+    // Note: visible history highlight is always reset at the beginning of the Selection Panel UI.
+
+    let should_display_visible_history = interacting_with_controls
+        || collapsing_response.header_response.hovered()
+        || collapsing_response
+            .body_response
+            .map_or(false, |r| r.hovered());
```

Comment:
Is it possible to make it so that the VH highlight stays visible when I'm editing the bound value text box, irrelevant of where my mouse cursor happens to be?

Reply:
Yes, it turns out. Excellent feedback!